src/StatusQ/Controls/StatusBaseButton.qml

```diff
@@ -19,131 +20,110 @@ Rectangle {
         Primary
     }
 
-    property StatusAssetSettings icon: StatusAssetSettings {
-        width: 24
-        height: 24
-    }
+    property StatusAssetSettings asset: StatusAssetSettings { }
 
     property bool loading: false
 
-    property alias hovered: sensor.containsMouse
+    property color normalColor
+    property color hoverColor
+    property color disabledColor
+
+    property color textColor
+    property color disabledTextColor
+    property color borderColor: "transparent"
 
     property int size: StatusBaseButton.Size.Large
     property int type: StatusBaseButton.Type.Normal
 
-    property alias text: label.text
-    property alias font: label.font
-
-    property real defaultLeftPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultRightPadding: size === StatusBaseButton.Size.Large ? 24 : 12
-    property real defaultTopPadding: {
-        switch (size) {
-            case StatusBaseButton.Size.Tiny:
-              return 5
-            case StatusBaseButton.Size.Small:
-              return 10
-            case StatusBaseButton.Size.Large:
-            default:
-              return 11
-        }
-    }
-    property real defaultBottomPadding: {
-        switch (size) {
+    property int defaultHorizontalPadding: (horizontalPadding > 0) ? horizontalPadding : ((size === StatusBaseButton.Size.Large) ? 24 : 12)
+    property int defaultVerticalPadding: {
+        if (verticalPadding > 0) {
+            verticalPadding;
+        } else {
+            switch (size) {
             case StatusBaseButton.Size.Tiny:
-              return 5
+                return 5;
             case StatusBaseButton.Size.Small:
-              return 10
+                return 10;
             case StatusBaseButton.Size.Large:
             default:
-              return 11
+                return 11;
+            }
```

Comment:
Just thinking... those "default" values, shouldn't they be readonly and hidden behind some d-pointer?

Reply:
Ah, could be then much simpler: just rename s/defaultHorizontalPadding/horizontalPadding and be done :) Same for vertical